### ISSUE TITLE
GpuIndexIVFPQ's quantizer can't use float16

### ISSUE THREAD

**Issue by GuodongTian** [facebookresearch/faiss#860]
# Summary

# Platform

OS: ubuntu 14.04.6 

Faiss version: 1.5.2

Faiss compilation options: 

Running on:
- [ ] CPU
- [x] GPU

Interface: 
- [x] C++
- [ ] Python

# Reproduction instructions
I defined an IndexIVFPQ object, and then obtained a GPU index using the function `faiss::gpu::index_cpu_to_gpu_multiple`, with the `GpuMultipleClonerOptions` defined by:
```
            faiss::gpu::GpuMultipleClonerOptions option;
            option.usePrecomputed = false;
            option.shard = true;
            option.useFloat16 = true;
            option.useFloat16CoarseQuantizer = true;
```
This gpu index can be successfully trained, but it crashes when adding vectors. Here is the log:
```
Faiss assertion '!quantizer_->getUseFloat16()' failed in int faiss::gpu::IVFPQ::classifyAndAddVectors(faiss::gpu::Tensor<float, 2, true>&, faiss::gpu::Tensor<long int, 1, true>&) at gpu/impl/IVFPQ.cu:124
``` 
The code at at gpu/impl/IVFPQ.cu:124 is:
```
FAISS_ASSERT(!quantizer_->getUseFloat16());
```
which means `useFloat16CoarseQuantizer` for GpuIndexIVFPQ is illegal.

Is there any plan to implement float16 qunantizer for GpuIndexIVFPQ?

**Comment by wickedfoo** [comment]
Sorry for this, I will work on it soon.

**Comment by GuodongTian** [comment]
> Sorry for this, I will work on it soon.

@wickedfoo thank you for your reply. I think it will be perfect if the centroids of PQ can also be saved as float16, because the they are highly memory consumed when nlist is large. I wander if this is possible.

**Comment by wickedfoo** [comment]
This is now fixed in the FB version and will work the next time we update the version on GitHub.

**Comment by mdouze** [comment]
Fixed in 1.6.3.
